code/datums/spells/reaper_lighter.dm

```diff
@@ -0,0 +1,85 @@
+/obj/effect/proc_holder/spell/reaper_lighter
+	name = "Reaper's Lighter"
+	desc = "Use this spell next to a vent to turn to smoke and crawl through it, even if the vents are welded. If you do not leave the vents within 15 seconds, you will be pulled back to where you started."
+
+	school = "transmutation"
+	charge_max = 15 SECONDS
+	clothes_req = FALSE
+	invocation = "none"
+	invocation_type = "none"
+	cooldown_min = 3 SECONDS //3 seconds reduction per rank
+	should_recharge_after_cast = FALSE //We start recharging manually once we leave the vent system
+	nonabstract_req = 1
+	sound = 'sound/items/zippolight.ogg'
+	action_icon_state = "lighter"
+
+	var/max_ventcrawl_time = 15 SECONDS
+	var/ventcrawl_warning_time = 10 SECONDS
+	var/time_to_leave = 0 //Set when cast, if you are still in vent by that time, you're pulled back
+	var/time_to_warn = 0 //Set when cast, warn the user they're going to leave soon.
+	var/warned = FALSE //So we dont repeat warnings over and over
+	var/in_progress = FALSE //Are we ventcrawling right now?
+	var/turf/starting_vent_turf
+
+/obj/effect/proc_holder/spell/reaper_lighter/create_new_targeting()
+	var/datum/spell_targeting/aoe/T = new()
+	T.allowed_type = /obj/machinery/atmospherics/unary
+	T.range = 1
+	T.use_turf_of_user = TRUE
+	return T
+
+/obj/effect/proc_holder/spell/reaper_lighter/valid_target(obj/machinery/atmospherics/unary/target, user)
+	return is_type_in_list(target, GLOB.ventcrawl_machinery)
+
+/obj/effect/proc_holder/spell/reaper_lighter/cast(list/targets, mob/user = usr)
+	var/obj/machinery/atmospherics/unary/target = targets[1]
+
+	if(!target)
+		charge_counter = charge_max
+		return FALSE
+	toggle_traits(user)
+	in_progress = TRUE
+	spawn_smoke(get_turf(target))
+	starting_vent_turf = get_turf(target)
+	user.handle_ventcrawl(target, TRUE)
+	time_to_leave = world.time + max_ventcrawl_time
+	time_to_warn = world.time + ventcrawl_warning_time
+	START_PROCESSING(SSfastprocess, src)
+
+/obj/effect/proc_holder/spell/reaper_lighter/process()
+	var/mob/user = action.owner
+	if(in_progress)
+		if(!warned && world.time > time_to_warn)
+			warned = TRUE
+			to_chat(user, "<span class='warning'>You will be pulled back to your entry vent soon!</span>")
+		if(world.time > time_to_leave)
+			toggle_traits(user)
+			in_progress = FALSE
+			warned = FALSE
+			start_recharge()
+			spawn_smoke(starting_vent_turf)
+			user.forceMove(starting_vent_turf)
+			starting_vent_turf = null
+	else
+		..()
```

Comment:
I think this logic would be better being part of a `status_effect`. rather than making use of the spells process. You could have the traits be applied when you add the status_effect and removed when it runs out. then you wouldn't need the `toggle_traits` proc. and you wouldn't need the `smoke_crawler` param you could instead check if the mob has the status effect.

Reply:
Do this please